app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/View.php

```diff
@@ -71,9 +71,8 @@ public function execute()
             $resultPage->getConfig()->getTitle()->prepend("#" . $shipment->getIncrementId());
             return $resultPage;
         } else {
-            $resultForward = $this->resultForwardFactory->create();
-            $resultForward->forward('noroute');
-            return $resultForward;
+            resultRedirect = $this->resultRedirectFactory->create();
+            return $resultRedirect->setPath('sales/shipment');
```

Comment:
And instead of using `} else {` block - surround `$shipment = $this->shipmentLoader->load()` with `try { } catch (NoSuchEntityException $e) {` and handle the error with redirect and adding `$this->messageManager->addErrorMessage(__('This shipment no longer exists.'));` there.